### PR TITLE
Fix macro missing from doc search

### LINKED ISSUE #51095: `panic`, `env`, and `vec` macros missing from std API doc search results
On stable (1.26.0), if I open the standard library API docs in my browser and search for one of the three strings `panic`, `env`, or `vec`, the search results include both the module and the macro by that name. Note that the module and the macro have the same path (`std::panic`, etc.).

On nightly 2018-05-25, those three macros no longer appear in search results, though their corresponding modules do. The macros still show up in the `std` crate index's macro list (`std/index.html#macros`). Since the macros and the modules have matching paths, it sounds like the macros are being shadowed somehow.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #51095.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by ollie27** [comment]
It's not just macros, it's any namespace collision like the following:

```rust
pub fn foo() {}
pub struct foo {}
```

only one `foo` appears in the search results.

**Comment by GuillaumeGomez** [comment]
I'll extend the fix then.

**Comment by GuillaumeGomez** [comment]
@ollie27: Your code doesn't work btw (for obvious reasons) but I extended my fix to make it work on all items at once.

**Comment by pietroalbini** [comment]
Ping from triage @QuietMisdreavus! This PR needs your review.

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -160,6 +160,7 @@
     // used for special search precedence
     var TY_PRIMITIVE = itemTypes.indexOf("primitive");
     var TY_KEYWORD = itemTypes.indexOf("keyword");
+    var TY_MACRO = itemTypes.indexOf("macro");
```
Comment (ollie27) [inline src/librustdoc/html/static/main.js]: `TY_MACRO` is no longer used anywhere.

**Comment by ollie27** [comment]
> Your code doesn't work btw (for obvious reasons)

What code?

This will need to be backported to beta as it affects 1.28.

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: True! Good catch!

**Comment by GuillaumeGomez** [comment]
This one:

```rust
pub fn foo() {}
pub struct foo {}
```

**Comment by GuillaumeGomez** [comment]
I'll just go ahead, this is a huge fix and I know that @QuietMisdreavus is very busy. Not having a fully working search is just too much of a problem.

@bors: r+ p=1

**Comment by pietroalbini** [comment]
Ping @rust-lang/rustdoc, can someone approve the beta backport (if you want to do that)? Thanks!

**Comment by GuillaumeGomez** [comment]
I approve it (if needed).

**Pull request merged**